components/bsx/Offer/MyOffer.vue

```diff
@@ -85,99 +82,155 @@
 
 import { Offer, OfferResponse } from './types'
 
-const components = {
-  Identity: () => import('@/components/identity/IdentityIndex.vue'),
-  Money: () => import('@/components/shared/format/Money.vue'),
-  NeoTable,
-  NeoTableColumn,
-  NeoSelect,
-  NeoButton,
+const { howAboutToExecute, initTransactionLoader, isLoading, status } =
+  useMetaTransaction()
+const { $apollo, $consola, $i18n } = useNuxtApp()
+const { urlPrefix, client } = usePrefix()
+const { accountId, isLogIn } = useAuth()
+
+const offers = ref<Offer[]>([])
+const destinationAddress = ref('')
+const selectedStatus = ref<AllOfferStatusType>(AllOfferStatusType.ALL)
+
+withDefaults(
+  defineProps<{
+    address: string
+    hideHeading: boolean
+  }>(),
+  {
+    address: '',
+    hideHeading: false,
+  }
+)
+
+const targetAddress = computed(() => destinationAddress || accountId)
+
+const getUniqType = () => {
+  const statusSet = new Set(offers.value.map((offer) => offer.status))
+  const singleEventList = Array.from(statusSet).map((type) => ({
+    type: type as AllOfferStatusType,
+    value: AllOfferStatusType[type],
+  }))
+
+  return [{ type: AllOfferStatusType.ALL, value: 'All' }, ...singleEventList]
 }
 
-@Component({
-  components,
-  filters: { formatDistanceToNow },
-})
-export default class MyOffer extends mixins(PrefixMixin, OfferMixin) {
-  public offers: Offer[] = []
-  public destinationAddress = ''
-  @Prop({ type: String, default: '' }) public address!: string
-  @Prop({ type: Boolean, default: false }) public hideHeading!: boolean
-
-  get targetAddress() {
-    return this.destinationAddress || this.accountId
+const fetchMyOffers = async () => {
+  if (!targetAddress.value) {
+    return
   }
 
-  mounted() {
-    if (this.targetAddress) {
-      this.$apollo.addSmartQuery<OfferResponse>('offers', {
-        client: this.urlPrefix,
-        query: acceptableOfferByCurrentOwner,
-        variables: { id: this.targetAddress },
-        manual: true,
-        result: ({ data }) => this.setResponse(data),
-        pollInterval: 10000,
-      })
-    }
+  try {
+    const { data } = await $apollo.query<OfferResponse>({
+      client: client.value,
+      query: acceptableOfferByCurrentOwner,
+      variables: {
+        id: targetAddress.value,
+      },
+    })
+    offers.value = data.offers
+  } catch (e) {
+    $consola.error(e)
   }
+}
 
-  public onClick = async (offer: Offer, withdraw: boolean) => {
-    const { caller, nft } = offer
-    const { id: collectionId, item } = tokenIdToRoute(nft.id)
-    this.isLoading = true
-    await this.submit(caller, item, collectionId, withdraw, this.fetchMyOffers)
+onMounted(() => {
+  if (targetAddress.value) {
+    // $apollo.addSmartQuery<OfferResponse>('offers', {
+    //   client: client.value,
+    //   query: acceptableOfferByCurrentOwner,
+    //   variables: { id: targetAddress.value },
+    //   manual: true,
+    //   result: ({ data }) => setResponse(data),
+    //   pollInterval: 10000,
+    // })
```

Comment:
how do you manage refreshing?

Reply:
GalleryItem should be a separate data island

Reply:
Introducing manual refresh button

![Screenshot 2023-08-10 at 10-24-08 KodaDot - NFT Market Explorer](https://github.com/kodadot/nft-gallery/assets/9987732/6a57fb1a-b9ac-4683-bfa1-346d968ae62b)